### ISSUE TITLE
DoneCMD not working in recent builds

### ISSUE THREAD

**Issue by dasweger** [boypt/simple-torrent#100]
Unable to get DoneCMD working on anything AFTER "build 1.2.3". "Build 1.2.4" added "donecmdThreshold", so I can only assume this is related.. but not seeing any other reports of this has me wondering if I'm missing something?

As mentioned above, DoneCMD working flawlessly in "build 1.2.3", DoneCMD.sh launching from local directory on "torrent" finished trigger. After using newer builds it wasn't working so I thought maybe my script was bad. So I tried using more basic scripts, changing trigger from "torrent" to "file" etc.. nothing worked. 

Finally looked in logs to see what was going on and I see that there are no "DoneCmd" logs happening after "StartTorrent"... so the DoneCMD is never being executed after a file or torrent finishes which explains why my DoneCMD.sh isn't working. DoneCmd is showing up properly in logs when using "build 1.2.3" and earlier. 

What I do not understand is why it's no longer working. Looking through the "donecmdThreshold" changes in build 1.2.4 I see no reason for things to stop working. 
-Am I missing something? 
-Does anything other than this line need to be in the config: "DoneCmd": "/root/doneCMD.sh",  ?
-Are other users having this issue on "build 1.2.4" or higher? 

I have no issues using "build 1.2.3" for now.. and I may look into building from source after reverting the "donecmdThreshold changes" but I'm curious and thought I'd at least ask here in hopes someone has any ideas. 
Thanks.

**Comment by boypt** [comment]
The threshold is that it won't call the cmd if the task finished in 30 seconds since the the task started.

I think if the task is actually finished but not stoped, then restart the process the task will finished "again", the threshold prevents this scenario.

But if your network speed is so fast that tasks will finished in 30s, that's a bug, maybe I should let the threshold configurable.

**Comment by dasweger** [comment]
Looks like that was exactly what was happening. With a 1gbit connection many well seeded files finish in under 30 seconds. 

I did see some files work with the donecmd in 1.2.4+ but then some weren't working so I had grabbed some really small files to do quick testing and they all finished within 10 seconds so they were under the 30-second threshold.  

It seems re-downloading already completed files in the same session also don't trigger the donecmd even when they take longer than 30 seconds? In my tests I had been either re-downloading the same file over and over.. or downloading very small files so in my limited testing nothing was triggering the donecmd. 

After looking over the code I had initially tried to add the threshold value into the config file but that unfortunately gave me an error. **Making the threshold configurable would be great.** 

After your response above I have tried some files that take longer than 30 seconds and can confirm the donecmd is now working... so just like you said, it was due to files finishing before 30 seconds (and also re-downloading files I had already completed in the same session). 

Appreciate the response and thanks for this great torrent app!
